src/http.rs

```diff
@@ -4,10 +4,17 @@ use std::time::Duration;
 
 use anyhow::Result;
 
+use crate::socks::Socks5Config;
+
 const HTTP_TIMEOUT: Duration = Duration::from_secs(30);
 
-pub(crate) fn get_client() -> Result<reqwest::Client> {
-    Ok(reqwest::ClientBuilder::new()
-        .timeout(HTTP_TIMEOUT)
-        .build()?)
+pub(crate) fn get_client(socks5_config: Option<Socks5Config>) -> Result<reqwest::Client> {
+    let builder = reqwest::ClientBuilder::new().timeout(HTTP_TIMEOUT);
+    let builder = if let Some(socks5_config) = socks5_config {
+        let proxy = reqwest::Proxy::all(socks5_config.to_url())?;
+        builder.proxy(proxy)
+    } else {
+        builder.no_proxy()
```

Comment:
Is not this the default and really needed here?

Reply:
But why do we need to disable the "system" proxy now? And maybe a comment on that is needed